### PR TITLE
[nn] implement `extend` method to sequential class

### PR DESCRIPTION
**Pull request by khushi-411**
Follows #71329

cc @kshitij12345 :)

### CODE REVIEW

**Review by albanD — COMMENTED** [review]
The change looks ok but I don't think the test does what you want.

File: test/test_nn.py [hunk 1]
```diff
@@ -1591,6 +1591,19 @@ def test_Sequential_append(self):
...
         self.assertEqual(nn.Sequential(l1).append(l2).append(l4), nn.Sequential(l1, l2, l4))
 
+    def test_Sequential_extend(self):
+        l1 = nn.Linear(10, 20)
+        l2 = nn.Linear(20, 30)
+        l3 = nn.Linear(30, 40)
+        l4 = nn.Linear(40, 50)
+        n1 = nn.Sequential(l1, l2)
+        n2 = nn.Sequential(l3, l4)
+        n3 = n1
```
[the remark this answers is not in the document]
Comment (albanD) [reply test/test_nn.py]: The test will always pass because `n1 is n3` so they are necessarily equal.

**Review by albanD — APPROVED** [review]
Thanks!

File: test/test_nn.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (khushi-411) [reply test/test_nn.py]: ah, right! just demonstrate this locally. Thanks, @albanD!

**Comment by khushi-411** [comment]
@albanD, I see one test failed. I'm not able to see the error log of this ([link](https://github.com/pytorch/pytorch/runs/7285653500?check_suite_focus=true)) failure. I guess it might be flaky. It shows:
Screenshot from 2022-07-12 00-03-11

What do you think? Thanks!

**Comment by khushi-411** [comment]
Hi, @albanD! Gentle ping :)

**Comment by khushi-411** [comment]
Hi, @albanD!

I think these errors are not-relatable. Rebasing it again to check.

Thanks ;)

**Pull request closed without merging**